SConstruct

```
@@ -9,6 +9,7 @@ libname = "EXTENSION-NAME"
 projectdir = "demo"
 
 localEnv = Environment(tools=["default"], PLATFORM="")
+localEnv["build_profile"] = "build_profile.json"
```

I'd suggest the following:

```suggestion
```python
if ARGUMENTS.get("build_profile", None) is None:
    # Using small build profiles decreases compile time. 
    # The file starts out empty. Edit it to benefit from decreased compile times.
    ARGUMENTS["build_profile"] = str("build_profile.json")
```
```

(further up at the top of the file)

`ARGUMENTS.setdefault("build_profile", "build_profile.json")` should also work on new SCons versions, but my local scons actually didn't support it so I'd suggest `if`.

The comment above can be improved with a link to a docs entry for build profiles once we have one.

---

Another option would be to fail if the `build_profile` option is manually set because it _probably_ shouldn't. But I don't see why I'd deny a user this configuration if it's explicitly requested.

---

On my SCons version this code makes help show:

```
build_profile: Path to a file containing a feature build profile ( /path/to/build_profile )
    default: None
    actual: build_profile.json
```

---

With the way it is it show default for me and can be replaced:
```
build_profile: Path to a file containing a feature build profile ( /path/to/build_profile )
    default: build_profile.json
    actual: copy.json
```

---

Is this the case for both implementations?

The reason I'd recommend against `localEnv["build_profile"] = "build_profile.json"` is because it would silently ignore `build_profile=some_build_profile.json`, overriding with the default value. This could potentially cause hard to debug problems.

---

Ah... perhaps this is because godot-cpp re-extracts the parameter after it's set?
I'm not sure which is the 'SCons' way to handle this. Yours seems better integrated, I suppose, so if it works it may be a better approach.